### PR TITLE
Added Event model and routes

### LINKED ISSUE #29: Event Schema and routes
### Description

Create event schema for storing event data in MongoDB with mongoose and relevant API routes for basic CRUD (Create-Read-Update-Delete) operations. Event data should have a title, a short description, a long description, an image path, event date, Google Calendar link and Google maps link.

### Concerning Folder

https://github.com/houseofgeeks/hg/tree/back-end/backend/models
https://github.com/houseofgeeks/hg/tree/back-end/backend/routes

### Screenshots/testimonials
<place your screenshots or testimonials here if applicable>

NA

### PR DESCRIPTION
**Pull request by ankiiitraj**
fixes #29 
<Do not delete this template>
Description: Added Events model and routes with CRUD operations

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes that were demanded in the linked issue.
- [ ] I've tested my code on a Chromium-based browser.
- [ ] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial
<provide a relevant screenshot or any testimonial for the solved issue if applicable>

### CODE REVIEW

File: backend/routes/event.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+const express = require("express");
+const router = express.Router();
+const auth = require("../util/auth");
+
+const Event = require("../models/event.model");
+
+router.get("/", (req, res, next) => {
```
Comment (EmperorYP7) [inline backend/routes/event.js:7]: We should be using `async` for this.

File: backend/routes/event.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  Event.find({})
```
Comment (EmperorYP7) [inline backend/routes/event.js:8]: Because this is to be awaited.

File: backend/routes/notice.js [hunk 3]
```diff
@@ -16,27 +17,32 @@ router.post("/", function (req, res, next) {
...
+      res.status(500).json({
+				message: 'Something went wrong'
+      });
       console.log(err);
     });
 });
 
 // Returns all the Notices (sorted : latest first)
 router.get("/", (req, res, next) => {
   Notice.find({})
```
Comment (EmperorYP7) [inline backend/routes/notice.js:34]: Wops, missed it! This should also be awaited.

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]
Although it doesn't seem to have any use now, we may come across problems if we don't use async functions. I think we should fix it everywhere. Otherwise, LGTM!

File: backend/routes/notice.js (hunk 3 above)
Comment (anktrj) [reply backend/routes/notice.js:34]: You do know right that this function returns a promise and .then is used when the promise is resolved and it is another way of writing async-await
suggestions?

File: backend/routes/event.js (hunk 1 above)
Comment (anktrj) [reply backend/routes/event.js:7]: this function is to be async right, so what's happening here is I am passing another function in arrow syntax as a callback, another way of writing async-await!

Comment (EmperorYP7) [reply backend/routes/event.js:7]: Alright, seems good to me then!

File: backend/routes/notice.js (hunk 3 above)
Comment (EmperorYP7) [reply backend/routes/notice.js:34]: Yep. All squared up! :wink:

**Review by EmperorYP7 — APPROVED** [review]
LGTM!

**Comment by ankiiitraj** [comment]
@tusharjain0022 - Review it! Then I will merge it

**Comment by tusharjain0022** [comment]
Everything seems good to me !! 🚀

**Pull request merged**